crates/iceberg-catalog/src/implementations/postgres/user.rs

```diff
@@ -0,0 +1,270 @@
+use super::dbutils::DBErrorHandler;
+use crate::api::management::v1::{
+    CreateRoleRequest, ListRolesResponse, ListUsersResponse, Role, UpdateUserRequest, User,
+    UserOrigin,
+};
+use crate::service::token_verification::UserId;
+use crate::service::{ErrorModel, Result};
+use sqlx::PgConnection;
+
+#[derive(sqlx::Type, Debug, Clone, Copy)]
+#[sqlx(rename_all = "kebab-case")]
+enum DbUserOrigin {
+    ExplicitViaRegisterCall,
+    ImplicitViaConfigCall,
+}
+#[derive(sqlx::FromRow, Debug)]
+pub struct UserRow {
+    id: String,
+    name: String,
+    email: Option<String>,
+    origin: DbUserOrigin,
+    created_at: chrono::DateTime<chrono::Utc>,
+    updated_at: Option<chrono::DateTime<chrono::Utc>>,
+}
+
+impl From<UserRow> for User {
+    fn from(
+        UserRow {
+            id,
+            name,
+            email,
+            origin,
+            created_at,
+            updated_at,
+        }: UserRow,
+    ) -> Self {
+        User {
+            id,
+            name,
+            email,
+            user_origin: match origin {
+                DbUserOrigin::ExplicitViaRegisterCall => UserOrigin::ExplicitViaRegisterCall,
+                DbUserOrigin::ImplicitViaConfigCall => UserOrigin::ImplicitViaConfigCall,
+            },
+            created_at,
+            updated_at,
+        }
+    }
+}
+
+pub(crate) async fn list_users<'e, 'c: 'e, E: sqlx::Executor<'c, Database = sqlx::Postgres>>(
+    connection: E,
+    include_deleted: bool,
+    name_filter: Option<&str>,
+) -> Result<ListUsersResponse> {
+    // TODO: impl pagination
+    let users = sqlx::query_as!(
+        UserRow,
+        r#"
+        SELECT
+            id,
+            name,
+            origin as "origin: DbUserOrigin",
+            email,
+            created_at,
+            updated_at
+        FROM users
+        where (deleted_at is null OR $1) AND (name ILIKE $2)
+        "#,
+        include_deleted,
+        name_filter.unwrap_or("%").to_string(),
+    )
+    .fetch_all(connection)
+    .await
+    .map_err(|e| e.into_error_model("Error fetching namespace".to_string()))?
+    .into_iter()
+    .map(User::from)
+    .collect();
+
+    Ok(ListUsersResponse {
+        users,
+        next_page_token: None,
+    })
+}
+
+pub(crate) async fn delete_user<'c, 'e: 'c, E: sqlx::Executor<'c, Database = sqlx::Postgres>>(
+    id: UserId,
+    connection: E,
+) -> Result<()> {
+    // TODO: @christian soft delete or hard delete? I'm leaning towards soft delete to avoid
```

Comment:
I am not sure - we must have a mechanism to delete users completely because of DSGVO.
So we need to find a way to implement it anyway. I tend towards hard delete or remove all PII data additionally - i.e. removing name and such.

Dangling references in AuthZ wouldn't be much of a problem.